src/YHttpWidgetsActionHandler.h

```diff
@@ -43,45 +49,106 @@ class YHttpWidgetsActionHandler : public YHttpHandler
     virtual void process_request(struct MHD_Connection* connection,
         const char* url, const char* method, const char* upload_data,
         size_t* upload_data_size, std::ostream& body, int& error_code,
-        std::string& content_encoding, bool *redraw);
-
-private:
+        std::string& content_type, bool *redraw);
 
+    int do_action( YWidget *widget, const std::string &action, struct MHD_Connection *connection, std::ostream& body );
 
     // TODO: move this somewhere else...
-
-    int do_action(YWidget *widget, const std::string &action, struct MHD_Connection *connection, std::ostream& body);
-
     template<typename T>
-    int action_handler(YWidget *widget, std::function<void (T*)> handler_func) {
+    int action_handler( YWidget *widget, std::ostream& body, std::function<void (T*)> handler_func ) {
         if (auto w = dynamic_cast<T*>(widget)) {
             try
             {
                 // allow changing only the enabled widgets, disabled ones
                 // cannot be changed by user from the UI, do not be more powerfull
-                if (handler_func && widget->isEnabled()) handler_func(w);
+                if( !widget->isEnabled() )
+                {
+                    std::ostringstream error;
```

Comment:
Um, do we really need `std::ostringstream` here? Simple `std::string` does not work here?

Reply:
I somehow decided to use stream here, as it performs better for concatenation. However we know the length of the string here, so you are right, we don't really need this here.
Thanks!